holidays/holiday_base.py

```diff
@@ -785,14 +785,23 @@ def _init_translation(self) -> None:
         """Initialize translation function based on language settings."""
         supported_languages = set(self.supported_languages)
         if self._entity_code is not None:
-            fallback = self.language not in supported_languages
-            languages = [self.language] if self.language in supported_languages else None
+            # Determine translation language: explicit setting first, then default.
+            # This logic is used to ensure that the language is set correctly for child entities
+            # in case they have a different language than their parent entity.
+            is_supported_language = self.language in supported_languages
+            if self.language and is_supported_language:
+                languages = [self.language]
+            elif self.default_language:
+                languages = [self.default_language]
+            else:
+                languages = None
```

Comment:
This may break the current translation behavior. I'll look into this later today (maybe we'll have to move this one into next release).